### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by gdumani**
# Created unit tests for 
- Books
- Persons
- Teachers
- Students
- Classrooms
- Rentals
- Decorators

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+require 'rspec'
+require_relative '../book'
+require_relative '../rental'
+require_relative '../person'
+require 'json'
+
+describe Book do
+  describe '#initialize' do
```
Comment (TadesseDev) [inline spec/book_spec.rb]: - [x] Hi Team, You did well testing your application. But we are required to have a test for all classes that you created while implementing the Decorator design pattern. This includes `Nameable` class and the `correct_name` method in it.
- [Optional] Also it's recommended to have one file per class. this includes a test for `class App` and `class Menu`. Please consider adding test for all.

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+require_relative '../person'
+require_relative '../book'
+require_relative '../rental'
+require_relative '../student'
+require_relative '../teacher'
+require 'json'
+
+describe Person do
+  describe '#initialize' do
+    it 'creates a new Person object' do
```
Comment (TadesseDev) [inline spec/person_spec.rb]: - [x] Dear Team, You did well organizing your test. But remember using `describe` and `context` is slightly different while organizing your test. `describe` is used to define the test group while `context` block can be used to provide additional context or conditions under which the tests are performed. Please use `context` inside `describe`. 
something like 
```ruby
describe Person do
  context "context here" do
    it "describe this specific test" do
    end
  end
end
```

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @banzaman and @banzaman 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
  - All changes requested have been implemented 🥇  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**